### PR TITLE
Taxonomy entity and endtpoints update

### PR DESCRIPTION
**Pull request by LBHSPreston**
This pull request is to implement additional updates to the taxonomies endpoint.
- The taxonomy response object that gets returned updated to the agreed endpoint specification https://app.swaggerhub.com/apis-docs/Nudge-Digital/fss-public-api/1.0.0#/taxonomy/getTaxonomy.
- A new 'description' field to be added to the taxonomies database table via migration as well as to the API entities.
- A new factory class has been added to standardise the conversion of the various taxonomy entities.
- A standard error response class has been added to standardise the error response.
- The StartUp class has been updated to ensure that API response objects meet the camelCase standard as specified in the API playbook.

This will bring the taxonomies endpoint in line with the API specifications.

### CODE REVIEW

File: LBHFSSPublicAPI/Startup.cs [hunk 1]
```diff
@@ -42,9 +42,6 @@ public static void ConfigureServices(IServiceCollection services)
                 .AddMvc(setupAction =>
                 {
                     setupAction.EnableEndpointRouting = false;
-                }).AddJsonOptions(jsonOptions =>
-                {
-                    jsonOptions.JsonSerializerOptions.PropertyNamingPolicy = null;
                 })
```
Comment (Duslerke) [inline LBHFSSPublicAPI/Startup.cs:45]: This was not mentioned in the description - should this change be there?

**Review by Duslerke — APPROVED** [review]
Look good.

File: LBHFSSPublicAPI/V1/Domain/TaxonomyEntity.cs [hunk 2]
```diff
@@ -6,7 +6,8 @@ public class TaxonomyEntity
     {
         public int Id { get; set; }
         public string Name { get; set; }
-        public int ParentId { get; set; }
+        public int? ParentId { get; set; }
+        public string Description { get; set; }
```
Comment (Duslerke) [inline LBHFSSPublicAPI/V1/Domain/TaxonomyEntity.cs:10]: Note: So Martin knows about the about the "Description" change, but did we tell there was a change to the ParentId to make it nullable? Does this matter to his schema?

Comment (LBHSPreston) [reply LBHFSSPublicAPI/V1/Domain/TaxonomyEntity.cs:10]: I don't think it matters in the schema.  The reason I made it nullable is that I don't think we expect all taxonomies to have a parent as there might be some who are top level taxonomies (no parent).

File: LBHFSSPublicAPI/Startup.cs (hunk 1 above)
Comment (Duslerke) [reply LBHFSSPublicAPI/Startup.cs:45]: Oh, never mind. Just noticed the last commit. So do I understand this correctly: does removing this, make json serializer go back to default options, which by default ensure that property names start with lower case?

Comment (LBHSPreston) [reply LBHFSSPublicAPI/Startup.cs:45]: Ahh I missed that.  I'll update the PR.

Comment (LBHSPreston) [reply LBHFSSPublicAPI/Startup.cs:45]: Yes you are correct.  It is not there by default.  It must have been added in while the API was being set up.  The default is camel case and not Pascal case as is used in the classes.

**Pull request merged**